gloo/transport/tcp/tls/openssl/dynamic_library.h

```diff
@@ -0,0 +1,16 @@
+struct DynamicLibrary {
+  DynamicLibrary(const DynamicLibrary&) = delete;
+
+  void operator=(const DynamicLibrary&) = delete;
+
+  DynamicLibrary(const char* name, const char* alt_name);
+
+  void* sym(const char* name);
+
+  ~DynamicLibrary();
+
+private:
+  const char* lib_name;
+  void* handle = nullptr;
+  const char* error = nullptr;
```

Comment:
Same as above: if you want to preserve an error message, explicitly copy it in to `std::string`

Reply:
addressed